lib/jobs/JobType.js

```diff
@@ -1,30 +1,50 @@
 import { Enum } from '@/lib/ClassUtils';
-import { ReportFormat, ReportType } from '@/lib/Constants';
+import { ReportExportMode, ReportFormat, ReportType } from '@/lib/Constants';
 import Joi from '@/lib/model/Joi';
+import CentrelineSelection from '@/lib/model/helpers/CentrelineSelection';
 
 class JobType extends Enum {
   get jobName() {
     return `${JobType.NAME_PREFIX}${this.name}`;
   }
-
-  async validateData(data) {
-    return this.dataSchema.validateAsync(data);
-  }
 }
 JobType.init({
   GENERATE_REPORTS: {
-    dataSchema: Joi.object().keys({
-      reports: Joi.array().items(
-        Joi.object({
-          type: Joi.enum().ofType(ReportType).required(),
-          id: Joi.string().required(),
-          format: Joi.enum().ofType(ReportFormat).required(),
-        }).unknown(),
-      ),
-    }),
+    getMetadata({ reportExportMode, s1 }) {
+      return { reportExportMode, s1 };
+    },
     getProgressTotal({ reports }) {
       return reports.length;
     },
+    async validateData(data) {
+      const dataSchema = Joi.object().keys({
+        ...CentrelineSelection,
+        reportExportMode: Joi.enum().ofType(ReportExportMode),
+        reports: Joi.array().items(
+          Joi.object({
+            type: Joi.enum().ofType(ReportType).required(),
+            id: Joi.string().required(),
+            format: Joi.enum().ofType(ReportFormat).required(),
+          }).unknown(),
+        ),
+      });
+
+      const { s1 } = data;
+      const dataNormalized = await dataSchema.validateAsync(data);
+      dataNormalized.s1 = s1;
```

Comment:
there's a few instances in this PR where we have to specially handle `CompositeId` values, all because our `Joi.compositeId()` extension changes the value from a string to an array.

just submitted this as #586 - we can address that separately at some point.  (it would also allow us to simplify the logic here: we could just define `dataSchema`, `metadataSchema` as properties and use those directly.)